r_shiny_tutorial.Rmd

```diff
@@ -0,0 +1,214 @@
+# R shiny tutorial
+
+Hanqin Zhou
+
+```{r include=FALSE}
+# this prevents package loading message from appearing in the rendered version of your problem set
+knitr::opts_chunk$set(warning = FALSE, message = FALSE)
+library(shiny)
+library(vcd)
+```
+
+## What is shiny?
+
+Shiny is an R package that makes it easy to build interactive web apps straight from R. You can host standalone apps on a webpage or embed them in R Markdown documents or build dashboards. You can also extend your Shiny apps with CSS themes, html widgets, and JavaScript actions. It is useful in data visualization because it allows users to interact with the data and help the analyzer to tell their story. This tutorial will help you to understand how shiny works, how to start your own shiny app and how to embed it with the visualization plots we learned in class! Sounds excited? Let's begin.
+
+## The architecture of a shiny app
+
+A shiny app is basically a web page which is maintained by a server running R script. Hence it has two parts. Part one is a UI theme which defines the styles and the interactions and part two is a Server which defines how displayed data changes when an action is triggered. To use it you must have the R package **shiny** installed and loaded in your code.
+
+![](resources/r_shiny_tutorial/1.png)
+
+## A simple shiny App
+
+The following is very simple shiny app template
+
+```{r eval = FALSE}
+ui <- fluidPage("Hi, it is a shiny app!") # start a shiny page
+
+server<- function(input,output){}
+
+shinyApp(ui = ui, server = server)
+
+```
+
+To get it rendered you just run the code above and you will get:
+
+![](resources/r_shiny_tutorial/2.png)
+
+Showing that this webpage is now running at your localhost. And you can visit this URL in a browser. And get something like this:
+
+![](resources/r_shiny_tutorial/3.png)
+
+To terminate the server just stop running the code.
+
+That's indicates your environment is set up! Then let's go further in to the inputs and outputs and see what we can play with shiny.
+
+### Inputs
+
+Input is the variable the app developer ask users to put in.
+
+![](resources/r_shiny_tutorial/6.png)
+
+Shiny supports many ways of input, the following is a quick snapshot of them and you can also discover more yourself on the [official website](https://shiny.rstudio.com/reference/shiny/1.7.0/).
```

Comment:
The line should contain an ! before the []. For example ![official website](https://shiny.rstudio.com/reference/shiny/1.7.0/) instead of [official website](https://shiny.rstudio.com/reference/shiny/1.7.0/)